### PR TITLE
DIT-973 Referred Days Elapsed Job

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/bindingtariffclassification/Module.scala [hunk 1]
```diff
@@ -16,30 +16,37 @@
...
       repositoryBinding
     )
   }
 
 }
+
+class ScheduledJobProvider @Inject()(activeDaysElapsedJob: ActiveDaysElapsedJob,
+                                     referredDaysElapsedJob: ReferredDaysElapsedJob)
+  extends Provider[ScheduledJobs] {
+  override def get(): ScheduledJobs = ScheduledJobs(Set(activeDaysElapsedJob, referredDaysElapsedJob))
```
Comment (ed0906) [inline app/uk/gov/hmrc/bindingtariffclassification/Module.scala]: Didnt seem to work with just `Set[ScheduledJob]` so I had to wrap it in a case class `ScheduledJobs` :(

File: conf/application.conf [hunk 2]
```diff
@@ -151,10 +151,15 @@ testMode = true
 upsert-permitted-agents = "binding-tariff-admin-frontend"
 
 scheduler {
-  days-elapsed {
+  active-days-elapsed {
```
Comment (ed0906) [inline conf/application.conf]: I've verified this isnt overridden in app-config-base/app-config-production

File: app/uk/gov/hmrc/bindingtariffclassification/scheduler/ReferredDaysElapsedJob.scala [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+      _ = Logger.info(s"$name: Updated Days Elapsed of Case [${c.reference}] from [${c.daysElapsed}] to [${referredDays.size}]")
+    } yield ()
+  }
+
+  private def toInstant: LocalDate => Instant = _.atStartOfDay(appConfig.clock.getZone).toInstant
+
+  private def bankHoliday(date: LocalDate)(implicit bankHolidays: Set[LocalDate]): Boolean = bankHolidays.contains(date)
+
+  private def weekend(date: LocalDate): Boolean = Set(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY).contains(date.getDayOfWeek)
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/bindingtariffclassification/scheduler/ReferredDaysElapsedJob.scala]: These three methods are duplicated in the existing job are they not?  Could we not extract somewhere

Comment (ed0906) [reply app/uk/gov/hmrc/bindingtariffclassification/scheduler/ReferredDaysElapsedJob.scala]: They could potentially go into the Scheduler Date Util

**Pull request merged**